doc/readme.html

```diff
@@ -288,6 +288,15 @@
 	  are usually found almost instantaneously, displaying the file
 	  name and line number of where the problem occurred.
 	</p>
+	<p>
+	  Debug mode, by default, uses the <code>-Og</code> flag. It promises most of the
+	  debugging experience of <code>-O0</code> but at a better performance.
+	  This is a reasonable choice for unit tests and enables numerous asserts
+	  within the library. Sometimes, however, one needs Debug mode to use
+	  <code>-O0</code> (e.g. GCC 6.2.0 optimizes out local variables with <code>-Og</code>).
```

Comment:
I guess not all users are familiar with what `-O0` actually means, so adding a description like "where all compiler optimizations are avoided and code and variables are exactly as indicated in the C++ program."